Respect CXX,CXXFLAGS, fix linking order

Patch written by Kacper Kowalik <xarthisius@gentoo.org>
--- a/makefiles/src/subdir.mk
+++ b/makefiles/src/subdir.mk
@@ -63,12 +63,14 @@
 ./src/undo.d \
 ./src/cl_support.d 
 
+CXX ?= g++
+CXXFLAGS ?= -O2 -ffast-math
 
 # Each subdirectory must supply rules for building sources it contributes
 src/%.o: ../src/%.cpp
 	@echo 'Building file: $<'
 	@echo 'Invoking: GCC C++ Compiler'
-	g++ -O2 -ffast-math -Wall -c -fmessage-length=0 `pkg-config --cflags gtk+-2.0 gthread-2.0;` -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)" -o"$@" "$<"
+	$(CXX) $(CXXFLAGS) -Wall -c -fmessage-length=0 `pkg-config --cflags gtk+-2.0 gthread-2.0;` -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)" -o"$@" "$<"
 	@echo 'Finished building: $<'
 	@echo ' '
 
--- a/makefiles/makefile
+++ b/makefiles/makefile
@@ -5,6 +5,7 @@
 -include ../makefile.init
 
 RM := rm -rf
+CXX ?= g++
 
 # All of the sources participating in the build are defined here
 -include sources.mk
@@ -44,7 +45,7 @@
 mandelbulber: $(OBJS) $(USER_OBJS)
 	@echo 'Building target: $@'
 	@echo 'Invoking: GCC C++ Linker'
-	g++ -L/usr/lib/x86_64-linux-gnu/ `pkg-config --libs gtk+-2.0 gthread-2.0` -o"mandelbulber"$(OBJS) $(USER_OBJS) $(LIBS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o"mandelbulber" $(OBJS) $(USER_OBJS) `pkg-config --libs gtk+-2.0 gthread-2.0` $(LIBS)
 	@echo 'Finished building target: $@'
 	@echo ' '
 
